Standards/CCSMB-3.md

```diff
@@ -0,0 +1,111 @@
+
+*Author: SkyTheCodeMaster*
+
+*Contributors: MasonGulu, spr2-dev*
+
+*Version: 1.0.0*
+
+*Last Revised: 2022-07-29*
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+bimg (abbreviation of "blit image") is a versatile image format for the ComputerCraft Minecraft mod.
+This specification aims to provide a standardized image format that can be easily implemented and used throughout the entire ComputerCraft ecosystem.
+
+# File Structure
+The examples shown throughout this specification are based on a sample image file, serialized using `textutils.serialize`.
+
+An image file MUST be separated into two sections: metadata data and frame data. The bimg parsers MUST be able to parse these two sections even if they are switched around. Meaning that they can be correctly written and read by parsers regardless of whether the metadata or the frame data comes first:
+![Format overview](media/CCSMB-3/file-structure.png?raw=true)
+
+Metadata contains information about the image itself, while the remaining of the file MUST contain one or more frames, representing the images.
+
+For more information on metadata, see [Metadata](#metadata). For more information on the image data and frames, see [Image Data](#image-data)
+.
+# Metadata
+bimg stores the following metadata along the image data in the file.
+
+| Field | Type | Description |
+|:--|:-:|:--|
+| `version` | `string` | Format version of the file. Used to differentiate two bimg file formats. Composed of three numbers separated by dots, defining the major, minor, and revision number respectively. |
+| `author` | `string` | Author of the image. |
+| `title` | `string` | Title of the image. Set by the author.
+| `description` | `string` | Description of the image. Set by the author. |
+| `creator` | `string` | Software that created the image. |
+| `date` | `string` | Date and time the image was created on, in ISO-8601 format. |
+| `width` | `number` | Width of the image in characters. |
+| `height` | `number` | Height of the image in characters. |
+| `animated` | `boolean` | Whether the file contains multiple frames and can be interpreted as an [animation](#animations).
+| (if animated) `secondsPerFrame` | `number` | Default length of each frame in seconds.
+| `palette` | `table` | Color palette table applied to the whole set of frames. For more information on palette formats, see [Palettes](#palettes).
+
+Renderers SHOULD support all metadata fields, and MUST support the `version` and `animated` fields.
+
+# Image Data
+Aside the metadata are stored the frames of the image. At least one frame MUST be present for the file to be considered valid.
+For animation support, see [Animations](#animations).
+
+## Frame Structure
+A single frame MUST be composed of multiple numerically-indexed tables, each representing a single line of the image.
```

Comment:
should this be multiple or at least one?

Reply:
I'd say at least one